src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -61,25 +71,29 @@ case class CsvRelation protected[spark] (
     val projection = schemaCaster(asAttributes(schema))
     val fieldNames = schema.fieldNames.toArray
 
-    val csvFormat = CSVFormat.DEFAULT
-      .withDelimiter(delimiter)
-      .withQuote(quote)
-      .withEscape(escape)
-      .withSkipHeaderRecord(false)
-      .withHeader(fieldNames: _*)
+    if(ParserLibs.isUnivocityLib(parserLib)) {
+      readCSV(baseRDD, fieldNames, schema.fields, projection, row)
+    } else {
+      val csvFormat = CSVFormat.DEFAULT
+        .withDelimiter(delimiter)
+        .withQuote(quote)
+        .withEscape(escape)
+        .withSkipHeaderRecord(false)
+        .withHeader(fieldNames: _*)
 
-    // If header is set, make sure firstLine is materialized before sending to executors.
-    val filterLine = if (useHeader) firstLine else null
+      // If header is set, make sure firstLine is materialized before sending to executors.
+      val filterLine = if (useHeader) firstLine else null
 
-    baseRDD.mapPartitions { iter =>
-      // When using header, any input line that equals firstLine is assumed to be header
-      val csvIter = if (useHeader) {
-        iter.filter(_ != filterLine)
```

Comment:
while attempting to fix broken unit tests i removed the optimization for this(drop the first line instead of filtering)...i will restore it as the unit test breakage was not due to that.